lib/cocoapods/generator/app_target_helper.rb

```diff
@@ -116,26 +113,23 @@ def self.add_swift_version(target, swift_version)
       # @param  [Symbol] platform
       #         the platform of the target. Can be `:ios` or `:osx`, etc.
       #
-      # @param  [Boolean] use_frameworks
-      #         whether to use frameworks or not when generating the contents of the import file.
-      #
       # @param  [String] name
       #         The name of the folder to use and save the generated main file.
       #
       # @return [Pathname] the new source file that was generated.
       #
-      def self.create_app_import_source_file(project, pod_target, platform, use_frameworks, name = 'App')
-        language = pod_target.uses_swift? ? :swift : :objc
+      def self.create_app_import_source_file(project, pod_target, platform, name = 'App')
+        language = [pod_target, *pod_target.recursive_dependent_targets].any?(&:uses_swift?) ? :swift : :objc
```

Comment:
We now check if the target itself or any of dependents use Swift.

Reply:
actually this can be a bit wrong as an ObjC pod will not be imported by the App unless it uses modules. 

The correct fix here is to just add a dummy swift file if any of the dependencies uses Swift but *not* the pod target itself.